Ar-Kareem/PythonDice
# Many major fixes and lots of tests

Major fixes:
- fix non responsive max func depth, not updating when settings is set
- fix broken clipping probs not updating sum

Good fixes:
- implement myand/myor and fix broken precedence
- support literal curly braces in f-string by properly escaping
- fix resolver correctly handling duplicate_arg_name
- implement special blankrv null

Test/Refactoring:
- implement global flag to remove zero probs in online response
- update README
- fix bug in logger.debug
- improve error description when myrange on RV

Added Numerous tests:
- test to check empty/None output
- 12 more failing tests (10 max_func_depth with autocast, and 2 dup)
- 2 more duplicate failing tests (duplicate var takes first only)
- more (skipped) failing tests for special max func depth and dupli…
- Add tests for special_none_type_object and Anydice_…
- add test for "&|" ops on ints
- more tests (binary &| ops on ints?)
- add test for curly brackets in output string
- add test for Seq with no params
- add failing test due to max_func_depth clashing with anydice casting
- add more detailed casting blank tests
- add real world failing test
